### PR TITLE
A12 - Wordle

### PR DESCRIPTION
**Pull request by vikrampg1710-tmm**

### CODE REVIEW

File: Assignments/Program.cs [hunk 1]
```diff
@@ -1,3 +1,202 @@
...
+   public string secretWord, text;
+   public int triesCount = 0, ithComment = 0;
+   public bool testing = false, gameOver;
+   public int originX, originY;               // Reference X & Y coordinates.
+   public List<List<char>> word = new (6);    // List To display and store user entered letters.
+   readonly List<char> grayList = new (),     // List to collect incorrect letters.
+                       blueList = new (),     // List to collect correct but misplaced letters.
+                       greenList = new ();    // List to collect correct and correctly placed letters.
+   public string[] dictWords, guessWords;     // To store loaded strings
+   public StreamWriter testOutputFile = null; // Created for testing purpose.
```
Comment (ravin-imm) [inline Assignments/Program.cs]: Why are these fields public? Moreover, naming conventions are not followed.

File: Assignments/Program.cs [hunk 2]
```diff
@@ -1,3 +1,202 @@
...
+         testOutputFile.Write (new string (' ', originX - 3));
+         testOutputFile.WriteLine (new string ('_', 31));
+         testOutputFile.WriteLine ();
+         testOutputFile.Write (new string (' ', originX - 5));
+         for (char c = 'A'; c <= 'Z'; c++) {
+            var (a, b) = greenList.Contains (c) ? ('{', '}')
+                        : blueList.Contains (c) ? ('[', ']')
+                        : grayList.Contains (c) ? ('(', ')')
+                        : (' ', ' ');
+            testOutputFile.Write ($" {a}{c}{b} ");
```
Comment (ravin-imm) [inline Assignments/Program.cs]: Why do we need a list of chars for each color here? Better way is to have a list of tuples where each tuple can carry the char and the color with which it has to be drawn.

File: Assignments/Program.cs [hunk 3]
```diff
@@ -1,3 +1,202 @@
...
+   /// <summary>Displays the updated game board</summary>
+   public void DisplayTheBoard () {
+      if (testing) {
+         // TYPING BOARD:
+         testOutputFile.Write (string.Join ("", Enumerable.Repeat ("\r\n", 3)));
+         if (text.Length == 1) testOutputFile.WriteLine ($"Try-{triesCount + 1}");
+         testOutputFile.Write (new string (' ', originX));
+         for (int i = 0; i < word.Count; i++) {
+            for (int j = 0; j < word[i].Count; j++) {
+               var (a, b, c) = (' ', ' ', word[i][j]);
```
Comment (ravin-imm) [inline Assignments/Program.cs]: These for loops are common in both conditions; for testing and while running the application. Can you simplify this?

**Review by ravin-imm — CHANGES_REQUESTED** [review]

File: Assignments/Program.cs (hunk 1 above)
Comment (vikrampg1710-tmm) [reply Assignments/Program.cs]: The reason for making the fields as public is to access those fields in the MSTest.

File: Assignments/Program.cs (hunk 2 above)
Comment (vikrampg1710-tmm) [reply Assignments/Program.cs]: I tried with Dictionary instead of tuple.

File: Assignments/Program.cs (hunk 1 above)
Comment (ravin-imm) [reply Assignments/Program.cs]: @vikrampg1710-tmm: Exposing all these fields is not the correct design. Think of a better way to do it. May be, in the testing mode some things like the secret word can be hardcoded etc.

File: Assignments/Program.cs [hunk 4]
```diff
@@ -1,3 +1,192 @@
...
+public class Wordle {
+   # region Properties --------------------------------------------------------
+   public string mSecretWord, mWord, mComment;
+   public bool mTesting = false, mGameOver;
+   int mX, mY;                                // // Reference X & Y coordinates
+   public int mTriesCount = 0;                // To count no.of tries made
+   public List<List<char>> mWords = new (6);  // List To store user entered letters
+   readonly Dictionary<char, ConsoleColor> mColorCode = new (); // To store foreground color to each char
+   public string[] mDictWords, mGuessWords;   // To store loaded strings
+   public StreamWriter testOutputFile = null; // Created for testing purpose
```
Comment (ravin-imm) [inline Assignments/Program.cs]: Exposing all these fields is not the correct design. Think of a better way to do it. May be, in the testing mode some things like the secret word can be hardcoded etc.

**Review by ravin-imm — CHANGES_REQUESTED** [review]

File: Assignments/Program.cs (hunk 4 above)
Comment (vikrampg1710-tmm) [reply Assignments/Program.cs]: Is there any way to access private fields outside the class, i.e. for testing purpose?

**Review by ravin-imm — APPROVED** [review]

**Pull request closed without merging**